python/graphstorm/model/hgt_encoder.py

```diff
@@ -252,17 +252,20 @@ def forward(self, g, h):
                     alpha = torch.sigmoid(self.skip[k])
                     if g.dstnodes[k].data.get('t') is not None:
                         t = g.dstnodes[k].data['t'].view(-1, self.out_dim)
-                        trans_out = self.drop(self.a_linears[k](t))
+                        trans_out = self.drop(t)
                         if g.is_block:
-                            trans_out = trans_out * alpha + h[k][:g.num_dst_nodes(k)] * (1-alpha)
+                            trans_out = trans_out * alpha + \
+                                self.a_linears[k](h[k][:g.num_dst_nodes(k)]) * (1-alpha)
                         else:
-                            trans_out = trans_out * alpha + h[k] * (1-alpha)
+                            trans_out = trans_out * alpha + self.a_linears[k](h[k]) * (1-alpha)
                     else:                       # Nodes not really in destination side.
                         warnings.warn("Warning. Graph convolution returned empty "
-                          f"dictionary, for node with type: {str(k)}")
-                        # So add psudo self-loop with feature copy.
-                        trans_out = self.drop(self.a_linears[k](h[k]))
-                        trans_out = trans_out * alpha + h[k] * (1-alpha)
+                          f"dictionary for node with type: {str(k)}. Pleaes check your data \
+                            for no in-degree {str(k)} nodes.")
+                        # So add psudo self-loop for the destination nodes with its own feature.
+                        dst_h = self.a_linears[k](h[k][:g.num_dst_nodes(k)])
```

Comment:
So the shape mis-match happens here?

Reply:
Yes. Original implementation does not take destination nodes as output.